tests/simulators/metadrive/test_metadrive.py

```diff
@@ -237,3 +337,56 @@ def test_duplicate_sensor_names(getMetadriveSimulator):
     assert img0.shape == (64, 64, 3)
     assert img1.shape == (64, 64, 3)
     assert not np.array_equal(img0, img1)
+
+
+def test_render_2D_saves_gif(getMetadriveSimulator, tmp_path):
+    outdir = tmp_path / "md_gifs"
+    outdir.mkdir()
+
+    simulator, openDrivePath, sumoPath = getMetadriveSimulator(
+        "Town01", render=True, render3D=False
+    )
+
+    simulator.screen_record = True
+    simulator.screen_record_filename = "render2d.gif"
+    simulator.screen_record_path = str(outdir)
```

Comment:
I'd rather use the supported API of `MetaDriveSimulator` than modify its internal attributes like this. Maybe you can have `getMetadriveSimulator` support arbitrary keyword arguments which get passed along to `MetaDriveSimulator` (rather than only `render` and `render3D`)?